baseplate/sidecars/trace_publisher.py

```diff
@@ -118,6 +123,26 @@ def publish(self, payload: SerializedBatch) -> None:
         )
 
 
+def build_and_publish_batch(
+    trace_queue: MessageQueue, batcher: TimeLimitedBatch, publisher: ZipkinPublisher, timeout: float
+) -> None:
+    while True:
+        message: Optional[bytes]
+
+        try:
+            message = trace_queue.get(timeout)
+        except TimedOutError:
+            message = None
+
+        try:
```

Comment:
This is the same logic as before, just moved into a helper function - it seems like None is still added to the batch. I can look into changing this behavior, but I think it might fall out of scope of this (already larger than expected) change.